### PR TITLE
Add and update random test data generators

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate Only updates to tests and test support.

### CODE REVIEW

File: test/random.go [hunk 1]
```diff
@@ -3,51 +3,234 @@ package test
...
+		minimum = RandomDurationMinimum()
+	}
+	if maximum > RandomDurationMaximum() {
+		maximum = RandomDurationMaximum()
+	}
+	return minimum + time.Duration(rand.Int63n(int64(maximum-minimum+1)))
+}
+
+func RandomDurationMaximum() time.Duration {
+	return 10 * 365 * 24 * time.Hour
```
Comment (jh-bate) [inline test/random.go]: they are all constants??

File: test/random.go [hunk 2]
```diff
@@ -3,51 +3,234 @@ package test
...
+	}
 	return minimum + rand.Intn(maximum-minimum+1)
 }
 
+func RandomIntMaximum() int {
+	return math.MaxInt32
+}
+
+func RandomIntMinimum() int {
+	return math.MinInt32
```
Comment (jh-bate) [inline test/random.go]: which is a constant??

File: test/random.go [hunk 3]
```diff
@@ -3,51 +3,234 @@ package test
...
+	}
+	return string(runes)
+}
+
+func RandomStringLengthMaximum() int {
+	return 64
+}
+
+func RandomStringLengthMinimum() int {
+	return 1
```
Comment (jh-bate) [inline test/random.go]: Naming??

File: test/random.go (hunk 1 above)
Comment (darinkrauss) [reply test/random.go]: Is is the maximum random duration. So the maximum duration that the random duration generators should generate. I name it this way for consistency and scoping reasons. That is everything starts with the prefix + "Maximum", "Minimum", etc...

File: test/random.go (hunk 2 above)
Comment (darinkrauss) [reply test/random.go]: See above...

File: test/random.go (hunk 3 above)
Comment (darinkrauss) [reply test/random.go]: See above...

File: test/random.go (hunk 1 above)
Comment (jh-bate) [reply test/random.go]: hmmm maybe in context :) I guess I keep thinking it needs a comment ... and then if it needs a comment it need renaming. anyway not blocking

**Comment by darinkrauss** [comment]
[Store device blobs](https://trello.com/c/bm3lPLwB/404-store-device-blobs)

**Pull request merged**